api/daemon.go

```diff
@@ -129,42 +129,48 @@ func checkForUpdate() (bool, string, error) {
 }
 
 // applyUpdate downloads and applies an update.
-func applyUpdate(version string) (err error) {
+func applyUpdate(version string) error {
 	manifest, err := fetchManifest(version)
 	if err != nil {
-		return
+		return err
 	}
 
 	// Get the executable directory.
 	binDir, err := osext.ExecutableFolder()
 	if err != nil {
-		return
+		return err
 	}
 
-	// Initialize the updater object.
-	up, err := update.New().VerifySignatureWithPEM([]byte(developerKey))
-	if err != nil {
-		// should never happen
-		return
-	}
+	// Configure the update.
+	opts := new(update.Options)
+	opts.SetPublicKeyPEM([]byte(developerKey))
 
 	// Perform updates as indicated by the manifest.
 	for _, file := range manifest[1:] {
+		// set update path
+		opts.TargetPath = filepath.Join(binDir, file)
+
 		// fetch the signature
-		var sig []byte
-		sig, err = getHTTP(version, file+".sig")
+		opts.Signature, err = getHTTP(version, file+".sig")
```

Comment:
no, I'm already spoofing the update server so it's about as easy as it can get. I would just need to create a valid update and store it on the fake server. Updates have worked in the wild, so we know that the signature verification works. It would still be good to have coverage for it, but I just haven't gotten around to it.

The same key is reused. Using a new key for each update is more secure, but it has some practical implications. For example, it prevents you from skipping an update, and it makes it harder for someone to verify that the key is a valid developer key (we would have to list all the keys somewhere).